feat(edge-services): update doc following migration (#6556)
scaleway/docs-content: pages/edge-services/how-to/configure-waf.mdx

Reviewer: ```suggestion
7. Click **Save changes** to exit Edit mode and save all your changes and deletions.
```

---
title: How to configure Edge Services Web Application Firewall

> description: Learn how to configure a Web Application Firewall (WAF) for Edge Services. Protect your backend from threats and malicious requests, and fine tune your settings to pick the right paranoia level and exclusions for your use case.

Reviewer 2: ```suggestion
description: Learn how to configure a Web Application Firewall (WAF) for Edge Services. Protect your backend from threats and malicious requests, and fine-tune your settings to pick the right paranoia level and exclusions for your use case.
```

import image2 from './assets/scaleway-edge-waf-popup.webp'
import image3 from './assets/scaleway-edge-waf-add-exc.webp'

> An Edge Services **W**eb **A**pplication **F**irewall (WAF) evaluates requests to your pipeline to determine whether they are potentially malicious. You can choose the [paranoia level](/edge-services/concepts/#paranoia-level) to be used when evaluating requests, and set [exclusions](/edge-services/concepts/#exclusions) to define traffic that shouldn't be filtered by WAF. Requests that are judged to be malicious are blocked or logged, depending on the settings you choose.

Reviewer 2: ```suggestion
An Edge Services **W**eb **A**pplication **F**irewall (WAF) evaluates requests to your pipeline to determine whether they are potentially malicious. You can choose the [paranoia level](/edge-services/concepts/#paranoia-level) to be used when evaluating requests, and set [exclusions](/edge-services/concepts/#exclusions) to define traffic that should not be filtered by WAF. Requests that are judged to be malicious are blocked or logged, depending on the settings you choose.
```

> An Edge Services **W**eb **A**pplication **F**irewall (WAF) evaluates requests to your pipeline to determine whether they are potentially malicious. You can choose the [paranoia level](/edge-services/concepts/#paranoia-level) to be used when evaluating requests, and set [exclusions](/edge-services/concepts/#exclusions) to define traffic that shouldn't be filtered by WAF. Requests that are judged to be malicious are blocked or logged, depending on the settings you choose.

Reviewer 3: ```suggestion
An Edge Services **W**eb **A**pplication **F**irewall (WAF) evaluates requests to your pipeline to determine whether they are potentially malicious. You can choose the [paranoia level](/edge-services/concepts/#paranoia-level) used when evaluating requests, and set [exclusions](/edge-services/concepts/#exclusions) to define traffic that shouldn't be filtered by WAF. Requests identified as malicious are blocked or logged depending on your settings.
```

This page walks you through the process of enabling and configuring WAF to protect your Edge Services pipeline. 

To read more about how WAF works, refer to our [Understanding WAF](/edge-services/reference-content/understanding-waf/) page.

<Requirements />

- A Scaleway account logged into the [console](https://console.scaleway.com)
- [Owner](/iam/concepts/#owner) status or [IAM permissions](/iam/concepts/#permission) allowing you to perform actions in the intended Organization
- An [Edge Services pipeline](/edge-services/how-to/create-pipeline/)

## How to enable and configure WAF

1. Click **Edge Services** in the **Network** section of the [Scaleway console](https://console.scaleway.com/) side menu. A listing of your pipelines displays.

2. Click the name of the pipeline you want to configure WAF for, then open the **Settings** tab.

> 2. In the **Web Application Firewall (WAF)** panel, click **Enable WAF**. A configuration pop-up displays.

Reviewer 3: ```suggestion
3. In the **Web Application Firewall (WAF)** panel, click **Enable WAF**. A configuration pop-up displays.
```

> 3. Choose the **paranoia level**, from 1 - 4, that is best adapted to your use case. The higher the paranoia level, the more sensitive WAF is to potential threats, and the more likely it is to classify a request as malicious. For help with choosing a paranoia level, see our [dedicated documentation](/edge-services/reference-content/understanding-waf/#waf-ruleset-and-paranoia-levels).

Reviewer 3: ```suggestion
4. Choose the **paranoia level**, from 1 to 4, that best suits your use case. The higher the paranoia level, the more sensitive WAF is to potential threats, and the more likely it is to classify a request as malicious. For help with choosing a paranoia level, see our [dedicated documentation](/edge-services/reference-content/understanding-waf/#waf-ruleset-and-paranoia-levels).
```

    <Message type="tip">
    After enabling WAF, you will be able to [set exclusions](#how-to-set-exclusions) that filter out requests matching certain criteria from being evaluated by WAF.
    </Message>

> 4. Select a WAF **mode**. Requests judged to be malicious can either be **blocked** and prevented from passing to the backend, or **logged** but allowed to pass.

Reviewer 3: ```suggestion
5. Select a WAF **mode**. Requests judged to be malicious can either be **blocked** and prevented from passing to the backend, or **logged** but allowed to pass.
```

> 5. Click **Save**.

Reviewer 3: ```suggestion
6. Click **Save**.
```

5. Click **Add** to add the exclusion.

6. Optionally, repeat steps 3 to 5 to add more exclusions (up to 100).

7. Click **Save changes** to save all the exclusions you added.

## How to edit exclusions

1. Click **Edge Services** in the **Network** section of the [Scaleway console](https://console.scaleway.com/) side menu. A listing of your pipelines displays.

2. Click the name of the desired pipeline, then open the  **Settings** tab.

2. In the WAF panel, click <Icon name="edit" /> next to the exclusion you want to edit.

3. Make edits to the filters as required. You cannot add more than one filter of each type (maximum of one path regex and one HTTP method filter per exclusion).

4. Click **Confirm** when you have finished editing.

5. Continue to edit or delete other exclusions as necessary.

6. Click **Save changes** to exit Edit mode and save all your changes.

## How to delete exclusions

1. Click **Edge Services** in the **Network** section of the [Scaleway console](https://console.scaleway.com/) side menu. A listing of your pipelines displays.

2. Click the name of the desired pipeline, then open the  **Settings** tab.

> 2. In the WAF panel, click **Edit exclusions** to enter Edit mode.

Reviewer 3: ```suggestion
3. In the WAF panel, click **Edit exclusions** to enter Edit mode.
```

> 3. Click the <Icon name="delete" /> icon next to the exclusion you want to delete. 

Reviewer 3: ```suggestion
4. Click the <Icon name="delete" /> icon next to the exclusion you want to delete. 
```

    A pop-up displays, asking you to confirm the deletion.

> 3. Click **Delete**.

Reviewer 3: ```suggestion
5. Click **Delete**.
```

> 4. Continue to edit or delete other exclusions as necessary.

Reviewer: ```suggestion
3. In the WAF panel, click **Edit exclusions** to enter Edit mode.

4. Click the <Icon name="delete" /> icon next to the exclusion you want to delete. 

    A pop-up displays, asking you to confirm the deletion.

5. Click **Delete**.

6. Continue to edit or delete other exclusions as necessary.
```

> 4. Continue to edit or delete other exclusions as necessary.

Reviewer 3: ```suggestion
6. Continue to edit or delete other exclusions as necessary.
```

You can edit WAF's paranoia level and mode (log or block) at any time.

1. Click **Edge Services** in the **Network** section of the [Scaleway console](https://console.scaleway.com/) side menu. A listing of your pipelines displays.

2. Click the name of the desired pipeline, then open the  **Settings** tab.

> 2. In the WAF panel, click <Icon name="edit" />.

Reviewer 3: ```suggestion
3. In the WAF panel, click <Icon name="edit" />.
```

> 3. Edit the paranoia level and mode as required.

Reviewer 3: ```suggestion
4. Edit the paranoia level and mode as required.
```

> 4. Click **Save**.

Reviewer: ```suggestion
3. In the WAF panel, click <Icon name="edit" />.

4. Edit the paranoia level and mode as required.

5. Click **Save**.
```

> 4. Click **Save**.

Reviewer 3: ```suggestion
5. Click **Save**.
```

    Your edits are saved, and you are returned to the Edge Services pipeline settings.

## How to disable WAF

You can disable WAF at any time.

1. Click **Edge Services** in the **Network** section of the [Scaleway console](https://console.scaleway.com/) side menu. A listing of your pipelines displays.

2. Click the name of the desired pipeline, then open the  **Settings** tab.

> 2. In the WAF panel, click **Disable WAF**. A confirmation pop-up displays.

Reviewer 3: ```suggestion
3. In the WAF panel, click **Disable WAF**. A confirmation pop-up displays.
```

> 3. Click **Disable** to confirm.

Reviewer: ```suggestion
3. In the WAF panel, click **Disable WAF**. A confirmation pop-up displays.

4. Click **Disable** to confirm.
```

> 3. Click **Disable** to confirm.

Reviewer 3: ```suggestion
4. Click **Disable** to confirm.
```
